integrations/github/src/haystack_integrations/components/connectors/github/issue_viewer.py

```diff
@@ -0,0 +1,212 @@
+import re
+from typing import Any, Dict, List, Optional
+
+import requests
+from haystack import Document, component, default_from_dict, default_to_dict, logging
+from haystack.utils import deserialize_secrets_inplace
+from haystack.utils.auth import Secret
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class GithubIssueViewer:
+    """
+    Fetches and parses GitHub issues into Haystack documents.
+
+    The component takes a GitHub issue URL and returns a list of documents where:
+    - First document contains the main issue content
+    - Subsequent documents contain the issue comments
+
+    ### Usage example
+    ```python
+    from haystack_integrations.components.connectors.github import GithubIssueViewer
+
+    viewer = GithubIssueViewer()
+    docs = viewer.run(
+        url="https://github.com/owner/repo/issues/123"
+    )["documents"]
+
+    print(docs)
+    ```
+    """
+
+    def __init__(
+        self,
+        github_token: Optional[Secret] = None,
+        raise_on_failure: bool = True,
+        retry_attempts: int = 2,
+    ):
+        """
+        Initialize the component.
+
+        :param github_token: GitHub personal access token for API authentication as a Secret
+        :param raise_on_failure: If True, raises exceptions on API errors
+        :param retry_attempts: Number of retry attempts for failed requests
+        """
+        self.github_token = github_token
+        self.raise_on_failure = raise_on_failure
+        self.retry_attempts = retry_attempts
+
+        # Only set the basic headers during initialization
+        self.headers = {
+            "Accept": "application/vnd.github.v3+json",
+            "User-Agent": "Haystack/GithubIssueViewer",
+        }
+
+    def _get_request_headers(self) -> dict:
+        """
+        Get headers with resolved token for the request.
+
+        :return: Dictionary of headers including authorization if token is present
+        """
```

Comment:
I like this pattern for an incidental reason. By not putting the resolve of the Secret in the init method we allow users to not need to provide the secret until run time which is nice especially when running pipeline validation like in the deepset platform where the secrets are not available to the pipeline in the pipeline builder. 

Could be nice to do this in the other components, basically to move the resolve_value() outside of the init method.

Reply:
Done for all components.